src/precice/impl/Participant.cpp

```diff
@@ -250,124 +217,56 @@ std::vector<MeshContext *> &Participant::usedMeshContexts()
   return _usedMeshContexts;
 }
 
-MeshContext &Participant::usedMeshContext(MeshID meshID)
-{
-  auto pos = std::find_if(_usedMeshContexts.begin(), _usedMeshContexts.end(),
-                          [meshID](MeshContext const *context) {
-                            return context->mesh->getID() == meshID;
-                          });
-  PRECICE_ASSERT(pos != _usedMeshContexts.end());
-  return **pos;
-}
-
-MeshContext const &Participant::usedMeshContext(MeshID meshID) const
-{
-  auto pos = std::find_if(_usedMeshContexts.begin(), _usedMeshContexts.end(),
-                          [meshID](MeshContext const *context) {
-                            return context->mesh->getID() == meshID;
-                          });
-  PRECICE_ASSERT(pos != _usedMeshContexts.end());
-  return **pos;
-}
-
-MeshContext &Participant::usedMeshContext(const std::string &name)
+MeshContext &Participant::usedMeshContext(std::string_view mesh)
 {
   auto pos = std::find_if(_usedMeshContexts.begin(), _usedMeshContexts.end(),
-                          [&name](MeshContext const *context) {
-                            return context->mesh->getName() == name;
+                          [mesh](MeshContext const *context) {
+                            return context->mesh->getName() == mesh;
                           });
   PRECICE_ASSERT(pos != _usedMeshContexts.end());
   return **pos;
 }
 
-MeshContext const &Participant::usedMeshContext(const std::string &name) const
+MeshContext const &Participant::usedMeshContext(std::string_view mesh) const
 {
   auto pos = std::find_if(_usedMeshContexts.begin(), _usedMeshContexts.end(),
-                          [&name](MeshContext const *context) {
-                            return context->mesh->getName() == name;
+                          [mesh](MeshContext const *context) {
+                            return context->mesh->getName() == mesh;
                           });
   PRECICE_ASSERT(pos != _usedMeshContexts.end());
   return **pos;
 }
 
-bool Participant::hasMesh(MeshID meshID) const
+bool Participant::hasMesh(std::string_view mesh) const
 {
-  return meshID < static_cast<int>(_meshContexts.size()) && _meshContexts.at(meshID) != nullptr;
+  return _meshContexts.count(mesh) > 0;
```

Comment:
I do not fully understand the conditional, but this comment is purely for a sanity check.

Reply:
The old condition was an index into a vector where values may be nullptr.
This is covered by the map now.